generic_config_updater/gu_common.py

```diff
@@ -147,12 +151,12 @@ def validate_config_db_config(self, config_db_as_json):
 
     def validate_field_operation(self, old_config, target_config):
         """
-        Some fields in ConfigDB are restricted and may not allow third-party addition, replacement, or removal. 
-        Because YANG only validates state and not transitions, this method helps to JsonPatch operations/transitions for the specified fields. 
+        Some fields in ConfigDB are restricted and may not allow third-party addition, replacement, or removal.
+        Because YANG only validates state and not transitions, this method helps to JsonPatch operations/transitions for the specified fields.
         """
         patch = jsonpatch.JsonPatch.from_diff(old_config, target_config)
-        
-        # illegal_operations_to_fields_map['remove'] yields a list of fields for which `remove` is an illegal operation 
+
+        # illegal_operations_to_fields_map['remove'] yields a list of fields for which `remove` is an illegal operation
```

Comment:
Please help ensure we test these specific scenarios on multi-asic as well (GCU blocks changes based on restrictions written into GCU, not reflected in yang models). Can be E2E if not UT

Reply:
Yes, that was tested as well, here is the output

```
admin@str3-7800-lc3-1:~/gcu$ sudo config apply-patch jsonpatch-illegal.json 
Patch: [{"op": "move", "path": "/asic0/PORTCHANNEL/PortChannel108/admin_status", "value": "down"}, {"op": "copy", "path": "/localhost/BGP_DEVICE_GLOBAL/STATE/tsa_enabled", "value": "true"}, {"op": "test", "path": "/asic0/BGP_DEVICE_GLOBAL/STATE/tsa_enabled", "value": "true"}]
('asic0', [{'op': 'move', 'path': '/PORTCHANNEL/PortChannel108/admin_status', 'value': 'down'}, {'op': 'test', 'path': '/BGP_DEVICE_GLOBAL/STATE/tsa_enabled', 'value': 'true'}])
NamSpace: asic0
generic_update_factory: <generic_config_updater.generic_updater.GenericUpdateFactory object at 0x7f78aa7ce7c0>
generic_update_factory: <generic_config_updater.generic_updater.GenericUpdateFactory object at 0x7f78aa7ce7c0>
Patch Applier: asic0: Patch application starting.
Patch Applier: asic0: Patch: [{"op": "move", "path": "/PORTCHANNEL/PortChannel108/admin_status", "value": "down"}, {"op": "test", "path": "/BGP_DEVICE_GLOBAL/STATE/tsa_enabled", "value": "true"}]
Patch Applier: Getting current config db.
Patch Applier: Simulating the target full config after applying the patch.
Failed to apply patch due to: The operation does not contain a 'from' member
Usage: config apply-patch [OPTIONS] PATCH_FILE_PATH
Try "config apply-patch -h" for help.

Error: The operation does not contain a 'from' member
```

Reply:
Yes, this has been tested as below output"

```
admin@str3-7800-lc3-1:~/gcu$ sudo config apply-patch jsonpatch-illegal.json
Patch: [{"op": "remove", "path": "/asic0/PFC_WD/GLOBAL/POLL_INTERVAL"}]
('asic0', [{'op': 'remove', 'path': '/PFC_WD/GLOBAL/POLL_INTERVAL'}])
NamSpace: asic0
generic_update_factory: <generic_config_updater.generic_updater.GenericUpdateFactory object at 0x7f5b40691940>
generic_update_factory: <generic_config_updater.generic_updater.GenericUpdateFactory object at 0x7f5b40691940>
Patch Applier: asic0: Patch application starting.
Patch Applier: asic0: Patch: [{"op": "remove", "path": "/PFC_WD/GLOBAL/POLL_INTERVAL"}]
Patch Applier: Getting current config db.
Patch Applier: Simulating the target full config after applying the patch.
Patch Applier: Validating all JsonPatch operations are permitted on the specified fields
Failed to apply patch due to: Given patch operation is invalid. Operation: remove is illegal on field: /PFC_WD/GLOBAL/POLL_INTERVAL
Usage: config apply-patch [OPTIONS] PATCH_FILE_PATH
Try "config apply-patch -h" for help.

Error: Given patch operation is invalid. Operation: remove is illegal on field: /PFC_WD/GLOBAL/POLL_INTERVAL
```